src/test/java/org/gridsuite/merge/orchestrator/server/BalancesAdjustmentServiceTest.java

```diff
@@ -40,19 +35,26 @@
     private UUID randomUuid3 = UUID.randomUUID();
 
     @Before
-    public void setUp() {
-        balancesAdjustmentService = new BalancesAdjustmentService(balancesAdjustmentServerRest);
+    public void setUp() throws IOException {
+        mockBackEnd = new MockWebServer();
+        mockBackEnd.start();
+        String baseUrl = String.format("http://localhost:%s", mockBackEnd.getPort());
+        balancesAdjustmentService = new BalancesAdjustmentService(baseUrl);
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        mockBackEnd.shutdown();
     }
 
     @Test
     public void test() {
-        when(balancesAdjustmentServerRest.exchange(anyString(),
-                eq(HttpMethod.PUT),
-                any(),
-                eq(String.class),
-                eq(randomUuid1.toString())))
-                .thenReturn(ResponseEntity.ok("{\"status\": \"SUCCESS\"}"));
-        String res = balancesAdjustmentService.doBalance(Arrays.asList(randomUuid1, randomUuid2, randomUuid3));
-        assertEquals("{\"status\": \"SUCCESS\"}", res);
+        mockBackEnd.enqueue(new MockResponse()
```

Comment:
@ChamseddineBhd you test doesn't test anything?  The answer of the mockserver is not mapped to a url query.
See study-server how it done with mock server.